### PR TITLE
Fix mediacodec resolution handling

### PR DESCRIPTION
**Pull request by eronnen**
* Fixes problem with specific codecs where configuring initial width and height is required
* Fixes the handling of scenarios where the codec changes the output width and height

### CODE REVIEW

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/codec/h264_mediacodec.c [hunk 1]
```diff
@@ -86,13 +87,18 @@ typedef AMediaFormat* (*AMediaCodec_getInputFormat_t)(AMediaCodec*);
 const int COLOR_FormatYUV420Planar = 19;
 const int COLOR_FormatYUV420Flexible = 0x7f420888;
 
+const int MEDIACODEC_MINIMUM_WIDTH = 320;
```
Comment (akallabeth) [inline libfreerdp/codec/h264_mediacodec.c:90]: if not used outside this file, make these `static` (same for the existing ones above)

File: libfreerdp/codec/h264_mediacodec.c [hunk 2]
```diff
@@ -86,13 +87,18 @@ typedef AMediaFormat* (*AMediaCodec_getInputFormat_t)(AMediaCodec*);
...
+const int MEDIACODEC_MINIMUM_HEIGHT = 240;
```
Comment (akallabeth) [inline libfreerdp/codec/h264_mediacodec.c:91]: As for the `320x240` is there a specific reason for this resolution or is the requirement just `!= 0` ? If the latter, better use `16` or similar as `FreeRDP` is used embedded where displays might be really small (h264 alignment is 16 if I recall correctly?)

**Review by akallabeth — COMMENTED** [review]
Overall LGTM, just not sure about the resolution (see comment)

**Comment by akallabeth** [comment]
@freerdp-bot test

File: libfreerdp/codec/h264_mediacodec.c (hunk 2 above)
Comment (eronnen) [reply libfreerdp/codec/h264_mediacodec.c:91]: According to MediaCodec documentation smaller resolutions than 320*240 are not supported:

https://developer.android.com/reference/android/media/MediaCodec#qualityFloor

With Java it's possible to use `MediaCodecInfo.VideoCapabilities` class (https://developer.android.com/reference/android/media/MediaCodecInfo.VideoCapabilities) in order to get the actual minimum/maximum resolutions supported by the codec but unfortunately this API is not exported to the NDK yet

Comment (akallabeth) [reply libfreerdp/codec/h264_mediacodec.c:91]: Ok, then we´ll just take this. Might be a good idea to reference the link with the constants.

**Review by akallabeth — APPROVED** [review]

**Comment by akallabeth** [comment]
@freerdp-bot test

**Pull request merged**